web/src/OverviewTable.test.tsx

```diff
@@ -235,134 +207,146 @@ describe("resource name filter", () => {
     })
 
     it("displays a `no matches` message if there are no matches", () => {
-      wrapper
-        .find(`${ResourceNameFilterTextField} input`)
-        .simulate("change", { target: { value: "eek no matches!" } })
-      wrapper.update()
+      let el = wrapper.querySelector(`${ResourceNameFilterTextField} input`)
+      userEvent.type(el, "eek no matches!")
 
-      expect(wrapper.find(TableNoMatchesFound)).toBeDefined()
+      expect(wrapper.querySelector(NoMatchesFound)).toBeDefined()
```

Comment:
this is a broader approach question that doesn't need to be figured out here, but i'm curious if you have thoughts! do you have any preference for querying the styled component element over text content or a11y semantics? (i actually didn't realize that we could query styled component elements with rtl till now! but i have liked the pattern of using semantic markup and dom content that it incentivizes.)

here's an example of what i mean by the latter:
```
let el = screen.queryByLabel("Filter resources by name")
userEvent.type(el, "eek no matches!")

expect(screen.findByText("No matches found")).toBeDefined()
```

it may be a half-dozen, six-of-another kind of question, or an opinion that we care about only when writing new tests.

Reply:
styled-components is MAGIC

basically a styled component creates a new, unique css class, and then they toString() to a selector for that css class, so you can use it in any querySelector.

i think there are pros and cons of both approaches. the industry goes through pendulum swings. the problem with searching by text is that if you have the text in more than one place in the DOM tree, you can end up testing the wrong thing by mistake.